File: flevents-backend/src/main/java/de/flyndre/fleventsbackend/dtos/OrganizationPreview.java
package de.flyndre.fleventsbackend.dtos;

import de.flyndre.fleventsbackend.Models.Role;
import lombok.Getter;
import lombok.Setter;

/**
 * This Class is the Data Transfer Object for the Preview of Organizations.
 * It provides getter as well as setter.
 * @implNote This DTO should only be returned in the Controller
 * @author Lukas Burkhardt
 * @version $I$
 */
@Setter
@Getter
public class OrganizationPreview {
    private String uuid;
    private String name;
    private String description;
    private String icon;
    private Role role;
    private String address;
    private String phoneContact;
    private String customerNumber;
}

File: flevents-frontend/src/models/organizationPreview.ts
import {Role} from "@/models/role";
import {MailConfig} from "@/models/mailConfig";

export interface OrganizationPreview {
  uuid: string,
  name: string;
  description: string;
  icon: string;
  role: Role;
  address: string;
  phoneContact: string;
  customerNumber: string;
  mailConfig: MailConfig;
}
